### PR TITLE
Testing: Store messages received and sent in RaftTestT

### PR DESCRIPTION
**Pull request by faineance**

### CODE REVIEW

File: test/RaftTestT.hs [hunk 1]
```diff
@@ -401,6 +411,7 @@ leaderElection' nid = do
 --------------------------------------------------------------------------------
 
 type TestNodeStatesConfig =  [(NodeId, Term, Entries StoreCmd)]
+type TestNodesResult = (TestNodeStates, TestEventsSeen)
```
Comment (faineance) [inline test/RaftTestT.hs]: This could have a better name

File: test/RaftTestT.hs [hunk 2]
```diff
@@ -71,6 +71,8 @@ instance Monad m => RaftStateMachine (RaftTestT m) Store StoreCmd where
 type TestEventChan m = TChan (STM m) (Event StoreCmd)
 type TestEventChans m = Map NodeId (TestEventChan m)
 
+type TestEventsSeen = [(NodeId, RPCMessage StoreCmd)]
```
Comment (tdietert) [inline test/RaftTestT.hs]: Can we rename to `TestRPCMsgEventsSeen`?

File: test/RaftTestT.hs [hunk 3]
```diff
@@ -89,6 +91,7 @@ data TestNodeEnv m = TestNodeEnv
   { testNodeEventChans :: TestEventChans m
   , testClientRespChans :: TestClientRespChans m
   , testNodeStates :: TVar (STM m) TestNodeStates
+  , testEventsSeen :: TVar (STM m) TestEventsSeen
```
Comment (tdietert) [inline test/RaftTestT.hs]: Same here, `testRPCMsgEventsSeen` for specificity.

File: test/RaftTestT.hs [hunk 4]
```diff
@@ -161,6 +164,9 @@ instance MonadConc m => RaftPersist (RaftTestT m) where
 
 instance forall m. MonadConc m => RaftSendRPC (RaftTestT m) StoreCmd where
   sendRPC nid rpc = do
+
+    testEventsSeenTVar <- asks testEventsSeen
+    atomically $ modifyTVar' testEventsSeenTVar $ ((nid, rpc) :)
```
Comment (tdietert) [inline test/RaftTestT.hs]: It seems to me that instead the name of this variable/field should be `testRPCMsgEventsSent`, not "seen". To me "seen" implies that these are the messages received, not the messages sent the node; Modifying the `TVar` here in `sendRPC` is collecting the RPC Message events sent.

File: test/RaftTestT.hs (hunk 1 above)
Comment (tdietert) [reply test/RaftTestT.hs]: yeah, but it's probably good for now ;)

**Pull request merged**